eigen3.sh

```
@@ -1,12 +1,20 @@
 package: Eigen3
 version: 3.4.0
 source: https://gitlab.com/libeigen/eigen.git
+prefer_system: .*
+prefer_system_check: |
+  EIGEN3_ROOT_EFF=${EIGEN3_ROOT:-${CMAKE_PREFIX_PATH%%:*}}
```

Shellcheck (0.11.0)

[info] Modification of EIGEN3_ROOT_EFF is local (to subshell caused by pipeline).

(SC2030)